src/llama.cpp

```diff
@@ -15581,6 +15842,16 @@ struct llm_tokenizer_bpe {
                     "\\p{N}",
                 };
                 break;
+            case LLAMA_VOCAB_PRE_TYPE_CHAMELEON:
+                regex_exprs = {
+                    "<sentinel:[0-9]+>",  // Sentinel tokens
+                    "(IMGIMG)((A|B|C|D|E|F|G|H|I){1,4})Z",  // Image tokens
+                    "([\t\n]|    |  )",  // directly from tokenizer.json
+                    "\\p{N}", // Individual digits
+                    "[\\p{P}\\$\\+<=>\\^~\\|`]+",  // Punctuation
```

Comment:
The proper regex for punctuation would need the `Po` category, which isn't supported until #8606. But for the ASCII-only part, HF `tokenizers` use Rust's [`is_ascii_punctuation`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_punctuation).

And I *think* `Isolated` means consecutive punctuation should be split (so no trailing `+`).

```suggestion
                    "[\\p{P}!-/:-@\\[-`{-~]",  // Punctuation, Isolated
```

(using ranges from the Rust documentation for conciseness and for completeness)